src/integrators/sdirk_integrators.jl

```diff
@@ -1,27 +1,29 @@
-function initialize!(integrator,cache::ImplicitEulerConstantCache,f=integrator.f)
+function initialize!(integrator, cache::ImplicitEulerConstantCache)
   integrator.kshortsize = 2
   integrator.k = eltype(integrator.sol.k)(integrator.kshortsize)
-  integrator.fsalfirst = f(integrator.t,integrator.uprev) # Pre-start fsal
+  integrator.fsalfirst = integrator.f(integrator.t, integrator.uprev) # Pre-start fsal
 
   # Avoid undefined entries if k is an array of arrays
   integrator.fsallast = zero(integrator.fsalfirst)
   integrator.k[1] = integrator.fsalfirst
   integrator.k[2] = integrator.fsallast
 end
 
-@muladd function perform_step!(integrator,cache::ImplicitEulerConstantCache,f=integrator.f)
-  @unpack t,dt,uprev,u = integrator
-  @unpack uf = cache
-  uf.t = t
+@muladd function perform_step!(integrator, cache::ImplicitEulerConstantCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f = integrator
+  @unpack uf,κ,tol = cache
+
+  κtol = κ*tol # save in cache instead of κ and tol?
```

Comment:
it's fine here. No need to over optimize. This allows for making the values adaptive in the future.